### PR TITLE
Add OneLocBuild to official build

### PR DESCRIPTION
**Pull request by jonfortescue**
Documentation on this [here](https://github.com/dotnet/arcade/blob/main/Documentation/OneLocBuild.md).

The changes to the eng/common files are bugfixes from later versions of arcade. When you upgrade to those versions there shouldn't be any merge conflicts as it will match what's already in the repo.

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -45,6 +45,13 @@ stages:
   displayName: Build and Test
 
   jobs:
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```
Comment (JoeRobich) [inline azure-pipelines-official.yml:48]: This check isn't strictly necessary, because this YML does not run for PRs.

**Review by JoeRobich — APPROVED** [review]

File: azure-pipelines-official.yml (hunk 1 above)
Comment (jonfortescue) [reply azure-pipelines-official.yml:48]: Ty for letting me know, will remove.

**Pull request merged**